### DISCUSSION TITLE
How upgrade to primeng 18 from previous version

### DISCUSSION THREAD

**Discussion by balagex** [primefaces/community#3149] [PrimeNG]
Hi,

As I see in the new primeng 18 documentation, there eill be a new configuration for the 18 version. Is there any upgrade guide, which will tell the way, how we should to set this configuration based on the existing project which using older primeng version. Which old configuration should keep, like styles in angular.json, and which should replace with new config, and how?

**Comment by djad442** [comment]
my suggestion would be to create a new project in a separate folder with ng new (version ng 18). then include the PrimeNG 18 libraries as per their start-up guide. Copy the src folder from your project to the new project and try.

**Reply by balagex** [reply]
Thanks for the suggestion, but I prefer to see an official step-by-step guide from the primeng developer side, because they exactly knows what and how had been changed behind the scenes.

**Reply by htdong** [reply]
I had been implementing v18 since it was first created till yesterday but still struggle with new theming approach. What I could share my experience is to recreate the entire new app as there are many breaking changes, especially if you use PrimeFlex or have many controls using templates.

**Reply by tkoseoglu** [reply]
I am in the same boat. The component and module changes are no issue. The new theming config, however, is. I had to abandon my upgrade to v18, unfortunately.

**Comment by NateRadebaugh** [comment]
@cagataycivici is there an official migration guide for how to migrate from a prime@17 to prime@18 theming engine? I have a large project with many components and theme customizations and am also trying to understand the recommended approach.

**Reply by NateRadebaugh** [reply]
Found the migration guide here that basically says the only way is to start from scratch for the new theming system:
https://primeng.org/guides/migration

**Comment by mhazak** [comment]
For anyone migrating a huge project:
Reserve more time for upgrading. I was updating from v17 to v18, but unfortunately, I was forced to give up due to so many breaking changes. 

A migration guide is not complete.

**Comment by DM-UOC** [comment]
It's so hard

So, If you want to upgrade, yes, its preference to create a new project to apply new features.

Y had to rollback the project, because de migration was so hard and new features doesn't apply correctly with the new changes.
